### PR TITLE
Add argon2 hash barrier

### PR DESCRIPTION
**Pull request by PlaceReporter99**
Extra protection!

### CODE REVIEW

File: flask_app.py [hunk 1]
```diff
@@ -135,8 +134,11 @@ def kill():
 
 @app.route("/update", methods=("POST",))
 def update():
+    argon_hash = argon2.PasswordHasher()
     key = request.headers.get("X-funky-password", "")
-    if compare_digest(sha256(key.encode()).hexdigest(), FUNKY_PASSWORD_HASH):
+    if compare_digest(
+        argon_hash.hash(sha256(key.encode()).hexdigest()), FUNKY_PASSWORD_HASH
```
[the remark this answers is not in the document]
Comment (PlaceReporter99) [reply flask_app.py:140]: @Lyxal would you be ok with this change, or not?

**Comment by Lyxal** [comment]
> Note that `FUNKY_PASSWORD_HASH` will have to be updated too. I assume @Lyxal has the original password, so he can do the argon hash thing to that directly instead of to the password hash, in which case you don't need `sha256` at all.

I uh don't.

That's pxeger's domain.

He set it up.

File: flask_app.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lyxal) [reply flask_app.py:140]: I don't see the necessity of this change. The hash is mostly to make sure that the update request comes from the right source. Pretty sure the server only uses it to make sure it's not wasting time and unnecessarily pulling changes.

**Review by Lyxal — CHANGES_REQUESTED** [review]
If it ain't broke don't fix it.

**Comment by Lyxal** [comment]
Wait hang on this doesn't require a new password at all, right? It's just hashed the previous hash and compares a double hash I think.

**Comment by Lyxal** [comment]
Because how can someone calculate a new hash value if they never knew the old password?

File: requirements_flask_app.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
 flask
 flask_cors
 poetry
+argon2-cffi
```
Comment (Lyxal) [inline requirements_flask_app.txt:4]: Before merging at all this'll need to be pip installed onto the server.

**Review by Lyxal — CHANGES_REQUESTED** [review]
Even if a new password isn't needed, I still need to prepare the server before merging.

File: requirements_flask_app.txt (hunk 2 above)
Comment (PlaceReporter99) [reply requirements_flask_app.txt:4]: Understood. Resolve this conversation when you are done with this.

Comment (Lyxal) [reply requirements_flask_app.txt:4]: it was already installed? interesting. well merging time I guess.

**Review by Lyxal — APPROVED** [review]

File: flask_app.py [hunk 3]
```diff
@@ -135,8 +134,11 @@
...
```
[the remark this answers is not in the document]
Comment (Lyxal) [reply flask_app.py:140]: wait hang on what's the salt?

**Review by Lyxal — CHANGES_REQUESTED** [review]

File: flask_app.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Lyxal) [reply flask_app.py:140]: Also, my testing shows that the hash is `$argon2id$v=19$m=65536,t=3,p=4$OaUMt+5zzknUWGILGyASCw$XHnqbPkGxhQ8IP/hKGscHb363bxBlmKkJI0nPG+P9sY`, not `$argon2id$v=19$m=65536,t=3,p=4$Je8jkzzOkWYdmb5SvdXeIQ$4c0zcj70kp0BXp5q7iJkb3CG+3vK5+5XpLx1UXRQkho`

**Review by Lyxal — CHANGES_REQUESTED** [review]

**Comment by PlaceReporter99** [comment]
@Lyxal must be using a random salt. do you want to salt with the word "cookie"?

**Comment by Lyxal** [comment]
Sure

**Comment by PlaceReporter99** [comment]
@Lyxal actually, argon2 says “cookie” is too short. How about “cookie cookie cookie cookie cookie”?

**Comment by Lyxal** [comment]
To be honest, I kinda don't see the need of using argon anyway. The previous hash will always be visible, and if you can manage to replicate it, then argon will hash the sha256 regardless of whether it's secure.

**Review by Lyxal — CHANGES_REQUESTED** [review]
I'll need to do some testing myself.

**Comment by Lyxal** [comment]
At this stage, I don't think this is really needed.

**Pull request closed without merging**